### ISSUE TITLE
[Security] Shield feature "Allow scripts once" doesn't work

### ISSUE THREAD

**Issue by palianskas** [brave/brave-browser#20503]
## Description 

*Allow scripts once* or *Allow* on individual scripts don't work, all scripts are still blocked 

## Steps to Reproduce

   1. Turn on *Scripts blocked* option in Shield
   2. Click *Allow scripts once* or *Allow* on individual scripts

## Actual result:

All scripts are still blocked

**Before** clicking *Allow* on _www.w3schools.com_ scripts
![image](https://user-images.githubusercontent.com/35391538/149490167-d11dd259-405f-4a6a-a674-962020c252fc.png)
**After** clicking *Allow* on _www.w3schools.com_ scripts
![image](https://user-images.githubusercontent.com/35391538/149490541-e7a292ed-539d-4f4a-abd4-1e16c8ecdb48.png)

## Expected result:
Currently blocked scripts get loaded if *Allow scripts once* was clicked or individual scripts get loaded if *Allow* was clicked

## Reproduces how often: 

Easily reproduced

## Brave version (brave://version info)

Brave | 1.34.80 Chromium: 97.0.4692.71 (Official Build) (64-bit)
Revision | adefa7837d02a07a604c1e6eff0b3a09422ab88d-refs/branch-heads/4692@{#1247}
OS | Windows 10 Version 21H1 (Build 19043.1415)

## Version/Channel Information:

- Can you reproduce this issue with the current release? 
  Yes
- Can you reproduce this issue with the beta channel? 
  Yes
- Can you reproduce this issue with the nightly channel? 
  Not tested

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? 
  Yes
- Does the issue resolve itself when disabling Brave Rewards? 
  No
- Is the issue reproducible on the latest version of Chrome? 
  Shield feature unavailable on Chrome

## Miscellaneous Information:

No additional info

**Comment by ryanbr** [comment]
Thanks for the report. Have chased down the regression in  v1.35.17    Looking into it further.


```
Nightly v1.35.27 (BAD)
Nightly v1.35.20 (BAD)
Nightly v1.35.17 (BAD)
Nightly v1.35.16 (GOOD)
Nightly v1.35.15 (GOOD)
Nightly v1.35.8  (GOOD)
```

**Comment by mariospr** [comment]
I'm afraid this is likely related to https://github.com/brave/brave-core/commit/6d7031c4ea7b461038e0ef5ce1eebcbf75175cc1 where, besides adapting tests, we were forced to adapt `UpdateContentSettingsToRendererFrames()`  in [browser/brave_shields/brave_shields_web_contents_observer.cc](https://github.com/brave/brave-core/blob/c1b6669/browser/brave_shields/brave_shields_web_contents_observer.cc#L67) to the removal of `WebContents::GetAllFrames()`.

@iefremov @bridiver If any of you could take a look at this as Brave Shields experts, that would be great. It's possible that the way I adapted  `UpdateContentSettingsToRendererFrames()` was not 100% correct in the end, but it's hard for me to know what exactly could be the problem here without deeper knowledge on how Shields work.

**Comment by diracdeltas** [comment]
"Allow scripts once" is a pretty useful feature in shields - could we get it added to either manual QA runs or as an automated test to make sure this regression doesn't happen again?

**Comment by RangeAntelope** [comment]
Greetings,
I'm using Brave Browser Version 1.34.80 Chromium: 97.0.4692.71 (Official Build) (64-bit) - Windows 10 Home.
Over the past few days when I try to use ALLOW SCRIPTS ONCE (and say there's 8 scripts wanting to execute) it just flips the screen and I have 8 scripts waiting to execute again (it does NOT allow them to execute one time as in the past).  When I've needed to access a site I don't want to give permanent JAVA Script executions to, I have to click the ALLOW button and then NoScript stops them all from flooding in and have execution permissions.
This is a hassle... ALLOW SCRIPTS ONCE used to work JUST ONCE and I didn't have to depend on NoScript as much.
I'm just wondering what might be going on?

**Comment by LaurenWags** [comment]
Added `QA/Blocked` label until this fix is uplifted to 1.35.x.

**Comment by goodov** [comment]
`Allow scripts once` doesn't work even in `release` currently. Reopening.

**Comment by goodov** [comment]
Apparently we don't track blocked embedded scripts. Seems like this was always an issue and if a website contains such script in the page, then we don't enable it on "Allow scripts once" action. Also iframes don't work with "Allow scripts once" action.

**Comment by kjozwiak** [comment]
> Apparently we don't track blocked embedded scripts. Seems like this was always an issue, and if a website contains such script in the page body, then we still don't allow it when "Allow scripts once" is clicked. Also iframes don't work with "Allow scripts once" action.

Closing this issue as https://github.com/brave/brave-core/pull/11928 is already associated with this issue. When QAing the above, I ran into https://github.com/brave/brave-browser/issues/20744. This will be fixed via https://github.com/brave/brave-core/pull/12023 and will most likely be uplifted along side this one.

**Comment by kjozwiak** [comment]
Removing `QA/Blocked` as this was uplifted into `1.35.x`. Should be checked with `1.35.97` or higher.

**Comment by stephendonner** [comment]
Verified `PASSED` using

Brave | 1.35.97 Chromium: 98.0.4758.72 (Official Build) (64-bit) 
--|--
Revision | d63e30f73734c5326037b0d94eceed77168d95e1-refs/branch-heads/4758@{#905}
OS | Linux

Steps:
1. new profile
2. launched Brave
3. via `brave://settings/shields`, enabled `Block scripts`
4. loaded `w3schools.com`
5. clicked on the Shields icon in the URL bar
6. noted there were 3 scripts blocked
7. clicked on `Allow scripts once`
8. confirmed script was unblocked
9. clicked again
10. confirmed last script was unblocked

scripts not blocked | `Block scripts` |  scripts blocked | unblocked using `Allow scripts once` | unblocked using `Allow scripts once`
----------------|------------------|------------------|---------------|-----------------
<img width="1438" alt="Screen Shot 2022-01-28 at 9 36 30 AM" src="https://user-images.githubusercontent.com/387249/151594886-1e4cc7f5-6358-4d83-9305-dbe0e2c0cfa3.png"> | <img width="1438" alt="Screen Shot 2022-01-28 at 9 29 49 AM" src="https://user-images.githubusercontent.com/387249/151594569-4a1a0715-5662-47eb-b73d-012f77d37e43.png"> | <img width="1438" alt="Screen Shot 2022-01-28 at 9 30 49 AM" src="https://user-images.githubusercontent.com/387249/151595003-77e62d66-7ab7-45e3-9c65-0a029bbcb8a0.png"> | <img width="1438" alt="Screen Shot 2022-01-28 at 9 30 57 AM" src="https://user-images.githubusercontent.com/387249/151595072-04483ad8-9b7c-4eba-b2ff-096ba228ccfd.png"> | <img width="1438" alt="Screen Shot 2022-01-28 at 9 31 06 AM" src="https://user-images.githubusercontent.com/387249/151595083-73f27dca-3727-411f-b8bb-c06e33dcd95b.png">

**Comment by RangeAntelope** [comment]
That looks very promising! I don't know if MAC OS using Brave will/would be effective using Brave Version 1.34.81 Chromium: 97.0.4692.99 (Official Build) (64-bit) which is the latest version according to ABOUT BRAVE.

I'm assuming that the ALLOW SCRIPTS ONCE function will be fixed in an upcoming 1.35.+ version so we'll see.
Thank you so much for attending to this so rapidly!
